### ISSUE TITLE
Warmup middleware

### ISSUE THREAD

**Issue by lmammino** [middyjs/middy#76]
In a discussion in #72, it seemed that middy users would benefit from having an implementation of a warm-up middleware.

The idea of warm-up helps with addressing the [cold start issue](https://serverless.com/blog/keep-your-lambdas-warm/) by just executing the lambda on a schedule every 5 or 10 minutes to make sure there is always at least one container bootstrapped.

When the lambda runs because of the warm-up schedule, it is not supposed to run business logic code (as that would possibly lead to undesired side effects), so it has to exit immediately.

The `warmupMiddleware` should run a `before` function that does basically 2 things:

  1. detect if the lambda is running because of the warm-up schedule by checking the event data
  2. if that's the case the middleware has to stop the execution of the lambda by completing the execution successfully and avoiding to run all the other middlewares.

Regarding point 1, we can have an option to define the check as a function that accepts the event as an argument and returns `true` if the execution is caused by the warm-up schedule and `false` otherwise. The default check can be compatible with [serverless-plugin-warmup](https://www.npmjs.com/package/serverless-plugin-warmup).

Regarding point 2, we might want to wait for #72 to be resolved so that we have an official "clean" way to stop the middleware execution and return early.

In any case, this is a potential implementation adapted from the one proposed by @DavidWells:

```javascript
module.exports = (config) => {
  const isWarmingUp = config.isWarmingUp || (event) => {
    return event.source === 'serverless-warmup-plugin'
  }
  return ({
    before: (handler, next) => {
      if (isWarmingUp(event)) {
          console.log('👋 Exiting early via warmUpMiddleware')
          return handler.context.succeed() // this will probably change when #72 is closed
      }
      next()
    }
  })
}
```

Thoughts?

**Comment by NPCRUS** [comment]
#81 Started a PR according to this issue

**Comment by lmammino** [comment]
Closed by #81
